### ISSUE TITLE
Possible to add classes within an emacs mode?

### ISSUE THREAD

**Issue by tanimislam** [purcell/mmm-mode#71]

**Comment by tanimislam** [comment]
In mmm-mode, can one define new classes within the emacs lisp file that defines a new custom mode rather than in the initializion file? That way, would I have to use the mmm-add-mode-ext-class function, or would I use other mmm-add functions?

**Comment by purcell** [comment]
Yes, this is possible. But `mmm-mode` settings do not generally belong with the definition of a major mode, because `mmm-mode`'s purpose is exactly to tie multiple major modes together. Do you have a concrete example to ask about?

**Comment by tanimislam** [comment]
Yes, in my case I would like to (at a high level) include major modes in simple code blocks with the beginning and ending syntax (like c_code...end_c_code) in my custom emacs mode. I spoke with <a href="https://github.com/jrblevin">Jason Blevins</a> and he mentioned copying from <a href="http://orgmode.org/">org-mode</a> (at a relatively low level) to define code blocks in markdown documents. However, I am not an emacs guru, so I cannot extract the bits of his code that can do this bit of functionality.

For reference, this is his <a href="http://jblevins.org/log/mmm">blog article</a> about using mmm-mode to initially define code blocks within markdown documents.
